### PR TITLE
MDWFP-29 Adding endpoint to get court events dates.

### PR DESCRIPTION
**Pull request by ldlharper**
Adding an endpoint to retrieve all court dates and result codes for a given prisoner. This data is required for the relevant remand calculation.

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/digitalwarrant/DigitalWarrantService.kt [hunk 1]
```diff
@@ -221,4 +222,30 @@ class DigitalWarrantService(
       ).id
     }
   }
+  @Transactional
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/digitalwarrant/DigitalWarrantService.kt:225]: The convention is to mark the service class as @Transactional(readonly=true) and override update/create functions with @Transactional

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/DigitalWarrantResource.java [hunk 2]
```diff
@@ -81,4 +85,15 @@ public Long createAdjustment(@PathVariable("bookingId") @Parameter(description =
                                   @RequestBody final Adjustment adjustment) {
         return digitalWarrantService.createAdjustment(bookingId, adjustment);
     }
+
+    @ApiResponses({
+        @ApiResponse(responseCode = "200", description = "The court date results.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = Integer.class))}),
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/DigitalWarrantResource.java:90]: this class doent look right

Comment (ldlharper) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/DigitalWarrantResource.java]: Done thanks.

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**